woocommerce/woocommerce
# Update workflow for rest api slack notifications

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:
Updates the test-assistant-api-rest-change-tracker workflow to auto-calculate milestone release dates instead of hardcoding





Closes https://github.com/woocommerce/woocommerce-quality/issues/793 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Fork this repo and update the workflow file changed in this PR
2. Add the secrets required for this PR : TEST_ASSISTANCE_BOT_TOKEN, WOO_CORE_REST_API_CHANGES_SLACK_CHANNEL, WOO_CORE_RELEASES_SLACK_CHANNEL
3. Add the correct value for TEST_ASSISTANCE_BOT_TOKEN (ask me if you don't know what it is)
4. Create a private slack channel for testing purposes. Call it whatever you want e.g. nigel-test-channel and obtain the Channel ID
5. Update the values of WOO_CORE_REST_API_CHANGES_SLACK_CHANNEL and WOO_CORE_RELEASES_SLACK_CHANNEL to the Channel ID created in step 4
6. Disable unnecessary workflows in your Fork to save bandwidth (you only need this workflow to be operational)
7. create and merge a PR in your Fork for testing purposes. If you have one already, you can use it.
8. Feel free to update the wait time in the workflow from 2m to make it operate more quickly
9. Assign it any valid Milestone with a value between 8.8.0 and 9.9.0
10. Add the label 'contains: rest api change' to your PR
11. Check that your test slack channel created in step 4 receives a message in the following format
![image](https://github.com/woocommerce/woocommerce/assets/105309450/4a12eed6-49b1-4305-ad54-90d4150f5760)
12. Remove the Milestone
13. Remove the label 'contains: rest api change'
14. Repeat steps 9 - 13 for other Milestones to test the PR




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
Update workflow to automatically calculate release milestones
</details>
